sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -957,6 +1007,13 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
   }
 
   Status = CmBuf->GetIndex(CmIndex);
+  if (PiESimdSurfaceMap->find((unsigned int)CmIndex->get_data()) !=
```

Comment:
Here and in other places - shouldn't accesses to `PiESimdSurfaceMap` be synchronized? Some threads may be creating surfaces and putting them into that map while this thread searches the map or modifies the map.

Reply:
SpinLock is applied for serialized access.